ui/components/MesheryPatterns.js

```diff
@@ -394,7 +400,13 @@ function MesheryPatterns({
     if (type === FILE_OPS.FILE_UPLOAD || type === FILE_OPS.URL_UPLOAD) {
       let body;
       if (type === FILE_OPS.FILE_UPLOAD) {
-        body = JSON.stringify({ pattern_data : { pattern_file : data }, save : true })
+        body = JSON.stringify({
+          pattern_data : {
+            name,
```

Comment:
@Abhishek-kumar09 what's the value for this 'name' key?

Reply:
is the actual value for this property kept 'undefined' for a valid reason?

Reply:
As far as I can see, `handlers/meshery_pattern_handler.go` checks for empty string and `handleSubmit` call passes either the file's name or a random name.. but we're never passing any of those values in line 405 for pattern_data object's 'name' key @Abhishek-kumar09

Reply:
Hi @MarioArriaga92,
Earlier the pattern name was not sent to the server, so your experience may have been due to that. 
If we are talking about the current scenario, when you upload a pattern with file, the file_name would be the pattern name and it will sent by the UI, or if you are creating your own pattern from the pattern configurator the random name like `meshery_23` would be your pattern name and will sent to server. When you are uploading the pattern from the URL, the name of the pattern is retrieved from the pattern_file sent to the server.
> .. but we're never assigning the value in line 405 for pattern_data object's 'name' property as the value

Actually, that's an es6 syntax, if your key and value variables are both having same name you can omit writing it.
e.g.
```
function f(name) {
  return {name};
}
```
is equivalent to:
```
function f(name) {
  return {name: name};
}
```

Reply:
Wow neat, thanks for clarifying @Abhishek-kumar09